node/compile-hook.js

```diff
@@ -11,10 +12,31 @@ if (! _compile.reified) {
   (Mp._compile = function (content, filename) {
     return _compile.call(
       this,
-      // Don't touch the file unless reification is enabled for the
-      // package that contains the file.
-      compile(content, filename),
+      compile(content, getOptions(filename)),
       filename
     );
   }).reified = _compile;
 }
+
+function getOptions(filename) {
+  const options = { filename: filename };
+  const mtime = mtimeOrNull(filename);
+
+  if (mtime !== null) {
+    options.cacheKey = {
+      source: "Module.prototype._compile",
```

Comment:
Just a salt to distinguish this call site from any other places where we might call `cachingCompiler.compile`.